frontend/src/components/Wallet.tsx

```diff
@@ -0,0 +1,47 @@
+import { Box, Center, Divider, HStack, Spinner, Text, VStack } from "@chakra-ui/react";
+import React from "react";
+import { WalletInfo, WalletInfoError, WalletInfoUpdate } from "./Types";
+
+interface WalletProps {
+    walletInfo: WalletInfo | null;
+}
+
+export default function Wallet(
+    {
+        walletInfo,
+    }: WalletProps,
+) {
+    let balance = <Spinner />;
+    let address = <Spinner />;
+
+    console.log(walletInfo);
+
+    if (walletInfo) {
+        if (walletInfo.type === "Error") {
```

Comment:
Not sure but maybe we want a dedicated error feed actually.

Reply:
Yeah, probably - not sure about that yet, but wanted to give this a try. Generally, dealing with the enum values is a bit of a pain (checking the type of the enum as string and then casting to payload is not optimal... - I did not have the brainpower to come up with something more elegant today ;)